modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -525,10 +547,10 @@ else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
 				(Predicate<T>)left, (Predicate<T>)right);
 		}
 		else {
-			return Optional.empty();
+			return null;
```

Comment:
@jeffwu0724 Only one branch in this `if-else-if` can be reached in a single method call, so you can make all branches directly return the predicate, and return `null` as the fallback.